tests/framework/extensions/supportmatrix/etcdversion.go

```diff
@@ -0,0 +1,26 @@
+package supportmatrix
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/sirupsen/logrus"
+)
+
+func CheckETCDVersion(client *rancher.Client, nodes []*nodes.Node, nodeRoles []string) error {
```

Comment:
Yeah I agree, simply just logging to me is not enough for this to be an extension. We should make automation validations without having to go through logs to manually validate.

Reply:
So we just check the version and log what it returns? So no validation?

Reply:
Sorry Markus, when I said functionality I was talking about this helper function's functionality. My point is, the output needs an extra step that somehow we validate that it's the version we want to have. While you're running the release testing, how do you decide that `etcd` version is correct? I am specifically talking about that part

Reply:
So depending on the k8s version you're just checking if the version has increased? Couldn't we automate that rather manually checking in the logs? So if it's 1.23-1.25 that version shouldn't change, but if it's 1.26 the version number should be higher?

Reply:
So that's the fun part lol. That isn't always a guarantee, at least in my experience. For instance, this past check, v1.26 had the same etcd version found in v1.25, but the flannel and canal images were updated. So in theory, you're right, but I guess that's the tricky part in terms of finding a validation that is robust and won't need constant upkeep

Reply:
Spoke offline regarding this. Agreed to return the values from the check calls, I will push this up in a new commit

Reply:
Resolved in latest commit.